chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/developer/attestation/DeveloperAttestationCheckInReportSummaryDataCollector.java

```diff
@@ -0,0 +1,56 @@
+package gov.healthit.chpl.scheduler.job.developer.attestation;
+
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+import gov.healthit.chpl.util.NullSafeEvaluator;
+import lombok.extern.log4j.Log4j2;
+
+@Component
+@Log4j2
+public class DeveloperAttestationCheckInReportSummaryDataCollector {
+    private static final String PENDING_DEVELOPER_ACTION = "Pending Developer Action";
+    private static final String PENDING_ACB_ACTION = "Pending ONC-ACB Action";
+    private static final String REJECTED = "Rejected";
+
+    public DeveloperAttestationCheckInReportSummary collect(List<DeveloperAttestationCheckInReport> developerAttestationCheckInReports) {
+        return DeveloperAttestationCheckInReportSummary.builder()
+                .developerCount(calculateDeveloperCount(developerAttestationCheckInReports))
+                .attestationsApprovedCount(calculateAttestationApprovedCount(developerAttestationCheckInReports))
+                .pendingAcbActionCount(calculatePendingAcbActionCount(developerAttestationCheckInReports))
+                .pendingDeveloperActionCount(calculatePendingDeveloperActionCount(developerAttestationCheckInReports))
+                .noSubmissionCount(calculateNoSubmissionCount(developerAttestationCheckInReports))
+                .build();
+    }
+
+    private Long calculateDeveloperCount(List<DeveloperAttestationCheckInReport> developerAttestationCheckInReports) {
+        return Long.valueOf(developerAttestationCheckInReports.size());
+    }
+
+    private Long calculateAttestationApprovedCount(List<DeveloperAttestationCheckInReport> developerAttestationCheckInReports) {
+        return developerAttestationCheckInReports.stream()
+                .filter(row -> row.getPublished())
+                .count();
+    }
+
+    private Long calculatePendingAcbActionCount(List<DeveloperAttestationCheckInReport> developerAttestationCheckInReports) {
+        return developerAttestationCheckInReports.stream()
+                .filter(row -> NullSafeEvaluator.eval(() -> row.getCurrentStatusName(), "").equals(PENDING_ACB_ACTION))
+                .count();
+    }
+
+    private Long calculatePendingDeveloperActionCount(List<DeveloperAttestationCheckInReport> developerAttestationCheckInReports) {
+        return developerAttestationCheckInReports.stream()
+                .filter(row -> NullSafeEvaluator.eval(() -> row.getCurrentStatusName(), "").equals(PENDING_DEVELOPER_ACTION))
+                .peek(x -> LOGGER.info(x.toString()))
+                .count();
+    }
+
+    private Long calculateNoSubmissionCount(List<DeveloperAttestationCheckInReport> developerAttestationCheckInReports) {
+        return developerAttestationCheckInReports.stream()
+                .filter(row -> NullSafeEvaluator.eval(() -> row.getCurrentStatusName(), "").equals(REJECTED)
```

Comment:
So the "no submission" count is any Developer that has no active CR, or the most recent CR has the status "REJECTED"? If the most recent one is REJECTED, and a new one is created, that will update the stats / checkInReports data to be in one of the PENDING states, right?

I'm trying to make sure I understand the logic. Assuming what I wrote above is correct, can you update the ticket checklist / requirements / test cases (if they exist) to make that clear?